java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -74,13 +89,39 @@
   private final CallBack callBack;
   private final List<BufferBacked> innerVectors;
 
-  public UnionVector(String name, BufferAllocator allocator, CallBack callBack) {
+  private int[] typeIds;
```

Comment:
@icexelloss, can you please provide insight into what you're trying to solve here?

Reply:
I believe this is about the format specification provides for fixed type code mapping, e.g.

```
type 0 -> 5
type 1 -> 10
type 2 -> 15
```

I think the original motivation was to be able to utilized fixed type codes for certain data types

In C++ you can see the `arrow::UnionType` constructor takes a vector of type codes https://github.com/apache/arrow/blob/master/cpp/src/arrow/type.h#L531

Reply:
Wes is correct. Before the change the Java class assumes the type id to be the int value of the enum MinorType, which breaks interoperability with C++.

Reply:
I haven't spent the time against understanding this yet but to clarify:

do you believe: 
- the java vector has an invalid representation?
or 
- the java vector assumes a certain possible subset of cases?

The fact that the java user layer uses a specific type mapping (rather than supporting every type mapping) isn't in itself a violation of the spec. Or am I missing something?

Reply:
The second. I believe the java vector assumes a certain possible subset of the cases - 

specifically the java vector assumes the typeIds are the ordinal value of the enum MinorType. If the java class is consuming an union vector that happens to using ordinal value of enum MinorType for typeIds (for instance, a union vector produced by java) then it would work. Otherwise it would not work.